admin-manual/security/csp-headers.rst

```diff
@@ -0,0 +1,263 @@
+   with the baseline CSP directive from AtoM 2.8's :ref:`config-app-yml`:
+
+   .. code-block:: none
+
+      Content-Security-Policy-Report-Only: default-src 'self'; font-src 'self'; img-src 'self' https://www.gravatar.com/avatar/ blob:; script-src 'self' 'nonce'; style-src 'self' 'nonce'; worker-src 'self' blob:; frame-ancestors 'self';
+
+   In app.yml this would look like:
+
+   .. code-block:: yaml
+
+      # Content Security Policy (CSP) header configuration. CSP settings apply
+      # only when a B5 theme is active, else these settings are bypassed.
+      csp:
+         response_header: Content-Security-Policy-Report-Only
+         directives: "default-src 'self'; font-src 'self'; img-src 'self' https://www.gravatar.com/avatar/ blob:; script-src 'self' 'nonce'; style-src 'self' 'nonce'; worker-src 'self' blob:; frame-ancestors 'self';"
+
+   For those who have crafted a custom theme, but haven't used AtoM's default Bootstrap 5 
+   Dominion theme as a foundation, it's recommended to start with a more restrictive 
+   `content-security-policy` header:
+
+   .. code-block:: none
+
+      Content-Security-Policy-Report-Only: default-src 'self'; font-src 'self'; img-src 'self'; script-src 'self'; style-src 'self'; frame-ancestors 'self';
+
+   In app.yml this would look like:
+
+   .. code-block:: yaml
+
+      # Content Security Policy (CSP) header configuration. CSP settings apply
+      # only when a B5 theme is active, else these settings are bypassed.
+      csp:
+         response_header: Content-Security-Policy-Report-Only
+         directives: "default-src 'self'; font-src 'self'; img-src 'self'; script-src 'self'; style-src 'self'; frame-ancestors 'self';"
+
+   These examples reference the ``Content-Security-Policy-Report-Only`` header -
+   this header will log violations in Chrome's Developer Tools but will not
+   prevent a script from loading and running. This is helpful when developing a
+   policy so that the pages and all scripts still load properly - while ensuring
+   all violations will be seen in the console log. When the policy definition is
+   ready to go live this CSP header type can be changed to ``Content-Security-Policy``
+   to activate enforcement.
+
+2. Monitor in the browser watching for CSP violations.
+
+   The line number can be obtained from the error as shown in Chrome's dev tools console.
+
+3. Look for and fix any violations by adding nonce to inline script, style, etc.
+
+   Use "view source" to find the implicated line - find and fix the violation in the
+   underlying code. Most of these are going to be inline assets - scripts, styles, etc.
+   See: :ref:`allow-inline-sources`.
+
+4. Activate policy enforcement.
+
+   Once you are convinced you've caught all the violations, activate enforcement
+   of the CSP policy by changing the header from
+   ``Content-Security-Policy-Report-Only`` to ``Content-Security-Policy``.
+
+.. _allow-inline-sources:
```

Comment:
i'd suggest prefacing this anchor with `csp-` so we can tell what the anchor relates to at a glance, and all CSP anchors are similarly named? (and generally all subheadings on this page, minus the page header one, which gets the `security-` prefix to match other pages in the section).... e.g. 

```
.. _csp-allow-inline-sources:
```

Reply:
Added